pkg/spi/aws.go

```diff
@@ -26,6 +28,29 @@ func (ms *PluginSPIImpl) NewSession(secret *corev1.Secret, region string) (*sess
 		Region: aws.String(region),
 	}
 
+	if workloadIdentityTokenFile, ok := secret.Data["workloadIdentityTokenFile"]; ok {
+		sess, err := session.NewSession()
+		if err != nil {
+			return nil, err
+		}
+		webIDProvider := stscreds.NewWebIdentityRoleProviderWithOptions(
+			sts.New(sess),
+			string(secret.Data["roleARN"]),
+			secret.Namespace,
+			stscreds.FetchTokenPath(workloadIdentityTokenFile),
+		)
+		creds, err := webIDProvider.Retrieve()
+		if err != nil {
+			return nil, err
+		}
+		cc := credentials.NewStaticCredentialsFromCreds(creds)
+		config := &aws.Config{
+			Region:      aws.String(region),
+			Credentials: cc,
+		}
```

Comment:
`config` is already defined, in that case it should be enough just to 

```suggestion
		config.Credentials = cc
```